services/shhext/service.go

```diff
@@ -179,6 +188,83 @@ func (s *Service) retrieveMessagesLoop(tick time.Duration, cancel <-chan struct{
 	}
 }
 
+type VerifyTransactionClient struct {
+	chainID *big.Int
+	url     string
+}
+
+func (c *VerifyTransactionClient) TransactionByHash(ctx context.Context, hash types.Hash) (coretypes.Message, bool, error) {
+	signer := gethtypes.NewEIP155Signer(c.chainID)
+	client, err := ethclient.Dial(c.url)
+	if err != nil {
+		return coretypes.Message{}, false, err
+	}
+
+	transaction, pending, err := client.TransactionByHash(ctx, commongethtypes.BytesToHash(hash.Bytes()))
+	if err != nil {
+		return coretypes.Message{}, false, err
+	}
+
+	message, err := transaction.AsMessage(signer)
+	if err != nil {
+		return coretypes.Message{}, false, err
+	}
+	from := types.BytesToAddress(message.From().Bytes())
+	to := types.BytesToAddress(message.To().Bytes())
+
+	return coretypes.NewMessage(
+		from,
+		&to,
+		message.Nonce(),
+		message.Value(),
+		message.Gas(),
+		message.GasPrice(),
+		message.Data(),
+		message.CheckNonce(),
+	), pending, nil
+}
+
+func (s *Service) verifyTransactionLoop(tick time.Duration, cancel <-chan struct{}) {
+	if s.config.VerifyTransactionURL == "" {
+		log.Warn("not starting transaction loop")
+		return
+	}
+
+	ticker := time.NewTicker(tick)
+	defer ticker.Stop()
+	client := &VerifyTransactionClient{
+		url:     s.config.VerifyTransactionURL,
+		chainID: big.NewInt(s.config.VerifyTransactionChainID),
+	}
+
+	for {
+		select {
+		case <-ticker.C:
+			accounts, err := s.accountsDB.GetAccounts()
+			if err != nil {
+				log.Error("failed to retrieve accounts", "err", err)
+			}
+			var wallets []types.Address
+			for _, account := range accounts {
+				if account.Wallet {
+					wallets = append(wallets, types.BytesToAddress(account.Address.Bytes()))
+				}
+			}
+
+			response, err := s.messenger.ValidateTransactions(client, wallets)
```

Comment:
It should take a context as an argument. The scenario is that the request fires but `cancel` channel is closed. The request should be interrupted and the loop should be exited immediately. Currently, it will wait for the request to finish.